smartfile/__init__.py

```diff
@@ -124,8 +129,32 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        return self.post('/path/oper/remove', path=deletefile)
+
+    def upload(self, filename, fileobj):
+        if filename.endswith('/'):
+            raise ValueError("File name should have no trailing slash")
```

Comment:
The file name should not have a trailing slash, so instead of raising an error I will remove the /